meta-ostro-xt/recipes-core/systemd/systemd_%.bbappend

```diff
@@ -0,0 +1,32 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
+
+SRC_URI_append = " \
+                  file://sleep.conf \
+                 "
+
+do_install_append() {
+
+    # Configure logind to suspend upon Power Key press. The desired (and
+    # the only possible on this platform) suspend state is 'freeze'. Be
+    # explicit about that setting by adding it in sleep.conf.
+    if ${@bb.utils.contains('PACKAGECONFIG', 'logind', 'true', 'false', d)}; then
+        install -m 0644 ${WORKDIR}/sleep.conf ${D}${sysconfdir}/systemd/
+        sed -i -e 's/.*\(HandlePowerKey=\).*/\1suspend/g' ${D}${sysconfdir}/systemd/logind.conf
+    fi
+
+    # systemd-logind checks whether /sys/power/state is writable but
+    # gets 'no' because the SMACK label it runs (System) is not allowed
+    # to write to floor label the file has. The process runs with a
+    # reduced capability set but we additionally need CAP_MAC_OVERRIDE to
+    # bypass SMACK MAC enforment to get suspend working.
+    #
+    # The actual write to /sys/power/state happens in systemd-sleep
+    # (from systemd-suspend.service) that runs unlimited capabilities.
+    if ${@bb.utils.contains('DISTRO_FEATURES', 'smack', 'true', 'false', d)}; then
```

Comment:
On Mon, Sep 12, 2016 at 11:14 AM, Patrick Ohly wrote:

> In meta-ostro-xt/recipes-core/systemd/systemd_%.bbappend
> <https://github.com/ostroproject/ostro-os-xt/pull/23#discussion_r78330225>
> :
>
> > +    # the only possible on this platform) suspend state is 'freeze'. Be
> > +    # explicit about that setting by adding it in sleep.conf.
> > +    if ${@bb.utils.contains('PACKAGECONFIG', 'logind', 'true', 'false', d)}; then
> > +        install -m 0644 ${WORKDIR}/sleep.conf ${D}${sysconfdir}/systemd/
> > +        sed -i -e 's/.*\(HandlePowerKey=\).*/\1suspend/g' ${D}${sysconfdir}/systemd/logind.conf
> > +    fi
> > +
> > +    # systemd-logind checks whether /sys/power/state is writable but
> > +    # gets 'no' because the SMACK label it runs (System) is not allowed
> > +    # to write to floor label the file has. The process runs with a
> > +    # reduced capability set but we additionally need CAP_MAC_OVERRIDE to
> > +    # bypass SMACK MAC enforment to get suspend working.
> > +    #
> > +    # The actual write to /sys/power/state happens in systemd-sleep
> > +    # (from systemd-suspend.service) that runs unlimited capabilities.
> > +    if ${@bb.utils.contains('DISTRO_FEATURES', 'smack', 'true', 'false', d)}; then
>
> I think that's the wrong place for the fix (or rather, workaround). It
> should be added to meta-intel-iot-security/meta-smack.
>

You mean move the same fix there? As we don't plan to pull new ostro-os for
now, I think a bbappend here is an OK approach.

> In addition, upstream systemd should get fixed to handle this better: if
> it needs to know whether systemd-sleep can write to a file, that test must
> be in systemd-sleep. At least a bug should be filed.
>
> You are right. I can take care of it. I was thinking of submitting a patch
to add ```CAP_MAC_OVERRIDE``` in upstream but a bug to re-think how the
check should be done is better.